sway-lib-std/src/auth.sw

```diff
@@ -192,6 +200,60 @@ pub fn caller_address() -> Result<Address, AuthError> {
     }
 }
 
+/// Get the owners of the inputs (of type `Input::Coin` or `Input::Message`) to a
+/// `TransactionScript`.
+///
+/// # Additional Information
+///
+/// The list is not deduplicated, so there may be repeated addresses in the returned vector.
+///
+/// # Returns
+///
+/// * [Vec<Address>] - The addresses of the owners of the inputs.
+///
+/// # Examples
+///
+/// ```sway
+/// use std::auth::caller_addresses;
+///
+/// fn foo(some_address: Address) {
+///     let addresses = caller_addresses();
+///
+///     assert(addresses.get(0).unwrap() == some_address);
+/// }
+/// ```
+pub fn caller_addresses() -> Vec<Address> {
```

Comment:
Shouldn't this return a `Result` like `caller_address()`?

I am also unsure how I feel about `caller_addresses()` as it's so close and makes it hard to differentiate between `caller_address()`.

Reply:
There is always atleast a single sender, so there is no need for a Result.

caller_addresses has a Result because it only returns a value if all the senders are the same, no such problem here.

I dont think the name being similar is a real problem as the return types are completely different, accidentally using one for another is unlikely

Reply:
Also IMO the similarity of the names shouldn't be an issue.

Reply:
It's more than one character of difference and a different type, I think confusion is unlikely.

I'm more concerned that `caller_address` doesn't have a name that conveys that it's only returning a unique one like `single_caller` and `all_callers` or something.